monai/apps/detection/utils/hard_negative_sampler.py

```diff
@@ -0,0 +1,300 @@
+
+        Args:
+            target_labels: list of labels per image.
+                For image i in the batch, target_labels[i] is a Tensor sized (A_i,),
+                where A_i is the number of samples in image i.
+                Positive samples have positive labels, negative samples have label 0.
+            concat_fg_probs: concatenated maximum foreground probability for all the images, sized (R,),
+                where R is the sum of all samples inside one batch, i.e., R = A_0 + A_1 + ...
+
+        Returns:
+            - list of binary mask for positive samples
+            - list of binary mask for negative samples
+
+        Example:
+            .. code-block:: python
+
+                sampler = HardNegativeSampler(
+                    select_sample_size_per_image=6, positive_fraction=0.5, min_neg=1, pool_size=2
+                )
+                # two images with different number of samples
+                target_labels = [ torch.tensor([0,1]), torch.tensor([1,0,2,1])]
+                concat_fg_probs = torch.rand(6)
+                pos_idx_list, neg_idx_list = sampler(target_labels, concat_fg_probs)
+        """
+        samples_per_image = [samples_in_image.shape[0] for samples_in_image in target_labels]
+        fg_probs = concat_fg_probs.split(samples_per_image, 0)
+        return self.select_samples_img_list(target_labels, fg_probs)
+
+    def select_samples_img_list(
+        self, target_labels: List[Tensor], fg_probs: List[Tensor]
+    ) -> Tuple[List[Tensor], List[Tensor]]:
+        """
+        Select positives and hard negatives from list samples per image.
+        Hard negative sampler will be applied to each image independently.
+
+        Args:
+            target_labels: list of labels per image.
+                For image i in the batch, target_labels[i] is a Tensor sized (A_i,),
+                where A_i is the number of samples in image i.
+                Positive samples have positive labels, negative samples have label 0.
+            fg_probs: list of maximum foreground probability per images,
+                For image i in the batch, target_labels[i] is a Tensor sized (A_i,),
+                where A_i is the number of samples in image i.
+
+        Returns:
+            - list of binary mask for positive samples
+            - list binary mask for negative samples
+
+        Example:
+            .. code-block:: python
+
+                sampler = HardNegativeSampler(
+                    select_sample_size_per_image=6, positive_fraction=0.5, min_neg=1, pool_size=2
+                )
+                # two images with different number of samples
+                target_labels = [ torch.tensor([0,1]), torch.tensor([1,0,2,1])]
+                fg_probs = [ torch.rand(2), torch.rand(4)]
+                pos_idx_list, neg_idx_list = sampler.select_samples_img_list(target_labels, fg_probs)
+        """
+        pos_idx = []
+        neg_idx = []
```

Comment:
nit: maybe asserting the same length could avoid some strange behaviour down the line